### PR TITLE
Add build workflow

### PR DESCRIPTION
**Pull request by longzheng**
Gitignore `dist`

Add build workflow for all push/pull requests.

Update publish workflow to use latest Node.

Update TypeScript version and fix type

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          registry-url: https://npm.pkg.github.com/
+          scope: "@121cast"
+      - name: Build
+        run: |
+          npm run build
```
Comment (stuartkeith) [inline .github/workflows/build.yml:17]: Why do we need build when the publish action builds?

Comment (longzheng) [reply .github/workflows/build.yml:17]: Just to validate commits/pull requests build without errors

**Review by stuartkeith — APPROVED** [review]

**Comment by stuartkeith** [comment]
This didn't seem to include the dist files, I tried to fix via https://github.com/121cast/XmpJS/commit/c51f2c2732a1680832b86082c8c2eaa1910b0283 but it didn't seem to work.

**Comment by stuartkeith** [comment]
Oh, I think that's the workflow failing, not necessarily the commit. Will look into it.

**Pull request merged**